### ISSUE TITLE
equalized-odds.md's COMPAS TPR/FPR table is fabricated - doesn't match this repo's actual COMPAS model

### ISSUE THREAD

**Issue by yakew7** [yakew7/Fair-Code#665]
**Where:** `explainers/equalized-odds.md`, lines 38, 46-50, 89-99 (the "Real-World Proof: COMPAS Analysis" section and its "Example output" code block).

**The gap:** the file claims "After training a classifier on the COMPAS dataset, we measured true positive rate (TPR) and false positive rate (FPR) broken down by race" and gives:
| Group | TPR | FPR |
|---|---|---|
| Black Defendants | 0.89 | 0.31 |
| White Defendants | 0.67 | 0.11 |

Training the repo's actual COMPAS model (same recipe as `COMPAS/unfair.py`: `Ethnic_Code_Text`-filtered, `ScoreText`-derived `is_high_risk` target, `RandomForestClassifier(random_state=42)`) and computing TPR/FPR by race on the held-out test set gives dramatically different numbers - independently reproduced twice, by two separate methods:

- Direct confusion-matrix reconstruction: Black TPR=0.926, FPR=0.791; White TPR=0.009, FPR=0.002
- Via `fairlearn.metrics.MetricFrame`: Black TPR=0.926, FPR=0.791; White TPR=0.009, FPR=0.002 (identical)

None of these resemble 0.89/0.31 (Black) or 0.67/0.11 (White) - especially FPR, which is off by roughly 2.5-20x. This is internally consistent with `disparate-impact.md`'s already-verified 87.16%/0.40% high-risk flag rates for this exact model (a model that flags 87% of Black defendants and 0.4% of White defendants necessarily has a huge FPR gap, not a modest 0.20 one).

**Repro:**
```python
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from fairlearn.metrics import MetricFrame, true_positive_rate, false_positive_rate

df = pd.read_csv('COMPAS/compas-scores-raw.csv')
df = df[df['Ethnic_Code_Text'].isin(['African-American', 'Caucasian'])]
df = df[df['DisplayText'] == 'Risk of Recidivism']
df['is_high_risk'] = df['ScoreText'].apply(lambda x: 1 if x in ['High', 'Medium'] else 0)
X = pd.get_dummies(df[['Sex_Code_Text', 'CustodyStatus', 'MaritalStatus']])
y = df['is_high_risk']
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
model = RandomForestClassifier(random_state=42)
model.fit(X_train, y_train)
y_pred = model.predict(X_test)
groups = df.loc[X_test.index, 'Ethnic_Code_Text']
frame = MetricFrame(metrics={'TPR': true_positive_rate, 'FPR': false_positive_rate}, y_true=y_test, y_pred=y_pred, sensitive_features=groups)
print(frame.by_group)
```
Output:
```
                   TPR       FPR
African-American  0.925643  0.791193
Caucasian         0.008584  0.001932
```

**Fix direction:** replace the table/"Example output" with real measured numbers, or clearly label them illustrative (as several sibling explainers already do for their COMPAS-adjacent examples).

**Comment by yakew7** [comment]
Correction to my own repro in this issue: the "real" comparison numbers I originally posted (Black TPR=0.926/FPR=0.791, White TPR=0.009/FPR=0.002) were actually computed on the biased model (with `race_binary` included as a feature), not the race-removed model the repro code shown here actually specifies - I mixed up which script I ran when filing this. Re-running the exact code block above (race excluded, `Sex_Code_Text`/`CustodyStatus`/`MaritalStatus` only) gives Black TPR=0.73/FPR=0.61, White TPR=0.64/FPR=0.43 - still nothing like the file's fabricated 0.89/0.31 vs 0.67/0.11 table, so the core finding holds, just with corrected real numbers. Fixed accordingly in the commit closing this issue, with an added caveat that this specific recipe showed real (not just last-digit) RandomForest output drift across two sessions on the same machine, so the numbers are labeled as this repo's reference-environment values rather than a universal constant.

**Comment by yakew7** [comment]
Follow-up correction to my own comment above: the "real (not just last-digit) RandomForest output drift across two sessions" claim I just made was wrong too - I re-checked, and it wasn't environmental drift at all. My original wrong numbers (0.926/0.791 Black, 0.009/0.002 White) exactly match this dataset's canonical **biased** model output (White predicted-positive rate 0.40%, the same well-established number from `COMPAS/unfair.py`'s 86.77% gap). I simply computed TPR/FPR on the race-included model both times, despite the repro code correctly excluding race. There's no cross-session model instability here - it was a repeatable mistake on my part, now fixed. Toned down the added caveat in the commit accordingly to the repo's standard "last digit can vary across BLAS backends" wording (matching reject-inference.md), rather than the stronger claim I made a moment ago.
